### ISSUE TITLE
Incorrect connections with multiple connections to the same switch

### ISSUE THREAD

**Issue by ajw1980** [netdisco/snmp-info#372]
I have a switch with these connections with lldp:
```
sw13# show lldp neighbors
Capability codes:
  (R) Router, (B) Bridge, (T) Telephone, (C) DOCSIS Cable Device
  (W) WLAN Access Point, (P) Repeater, (S) Station, (O) Other
Device ID             Local Intf      Hold-time  Capability  Port ID
sw05            Eth1/3          120       BR          Ethernet1/3
sw05            Eth1/45         120       BR          Ethernet1/1
```
In the netdisco web interface, sw13 show this:
sw13 port Eth1/3 -> sw05 Ethernet1/1
sw13 port Eth1/45 -> no connection

I checked the snmp output and it seems correct as well. Let me know what type of debug output you need for this.

**Comment by inphobia** [comment]
after private discussion it seems nxos6 returns ifAlias names instead of ifDescr for lldpLocPortDesc.

lldpLocPortDesc states: "If the local agent supports IETF RFC 2863, lldpLocPortDesc object should  have the same value of ifDescr object".

in nxos6 those values can be found in lldpLocPortId however.

i think the cleanest solution is to change
https://github.com/netdisco/snmp-info/blob/a85c7a0404d2dc8b05f3be1c8ff1e33d1645754a/lib/SNMP/Info/LLDP.pm#L149
lldpLocPortDesc to a function, and then overwrite it for nxos6 in the Nexus.pm module.

patch will come in a few hours.

**Comment by inphobia** [comment]
https://github.com/inphobia/snmp-info/tree/nn_nxos6_lldpfix has most of the framework and i'm currently testing this for regressions.

the only part that still needs to be done is modifying lldp_lport_desc() in Nexus.pm to make nxos6 return lldpLocPortId there. shouldn't take to long.

**Comment by inphobia** [comment]
@netdisco/netdisco-admins  and @netdisco/snmp-info-admins 
there are many ways to fix this, but to me the safest way for this issue seems how i did it here, using oid's directly is frowned upon anyway, and perhaps other lldp fixes can make use of this too.

since lldp is such a core module i'm hestitant to make any other changes.

as always all feedback welcome.

**Comment by inphobia** [comment]
any change to check if https://github.com/inphobia/snmp-info/tree/nn_nxos6_lldpfix fixes your issue?

**Comment by ajw1980** [comment]
Looks ok to me. I applied that patch and I am getting the correct connections.

**Comment by inphobia** [comment]
pull request #375 is created, would like a few more eyeballs but will merge into master in a week or so.

**Comment by ajw1980** [comment]
Any update on getting the pull request merged?

**Comment by inphobia** [comment]
due to personal stuff this got sidelined. i expect it to be included in snmp::info 3.71 when it gets released.

**Comment by inphobia** [comment]
just committed this, finally.

the next snmp::info release will contain the fix.
